app/src/main/java/ch/sdp/vibester/activity/LyricsBelongGameActivity.kt

```diff
@@ -64,30 +102,50 @@ class LyricsBelongGameActivity : AppCompatActivity() {
     }
 
     /**
-     * check if the given string belongs to the lyrics of the song
+     * get the lyrics of a given song
      */
-    private fun checkLyrics(lyricToBeCheck: String) {
-        val service = ServiceBuilder.buildService(baseUrl,LyricsOVHApiInterface::class.java)
-        val call = service.getLyrics("Imagine Dragons", "Thunder")
+    private fun getAndCheckLyrics(songName: String, artistName: String, speechInput: String) {
+        val service = ServiceBuilder.buildService(baseUrlLyrics, LyricsOVHApiInterface::class.java)
+        val call = service.getLyrics(artistName, songName)
         call.enqueue(object: Callback<Lyric> {
             override fun onFailure(call: Call<Lyric>?, t: Throwable?) {}
 
             override fun onResponse(call: Call<Lyric>?, response: Response<Lyric>?) {
+
                 if (response != null) {
-                    val originLyric = response.body().lyrics.toString()
-                    findViewById<TextView>(R.id.lyricMatchResult).text = if (originLyric.contains(lyricToBeCheck, ignoreCase = true)) "res: correct" else "res: too bad"
+                    val result = response.body()
+                    if (response.isSuccessful && result != null) {
+                        lyrics = result.lyrics.toString().replace(",", "")
+                        checkLyrics(speechInput, lyrics) // be sure the lyrics is ready when checking
+                    } else {
+                        findViewById<TextView>(R.id.lyricMatchResult).text = "No lyrics found, try another song"
+                    }
                 }
             }
         })
     }
 
-    // helper functions to test private functions
+    /**
+     * show the result of lyrics matching
+     */
+    private fun checkLyrics(lyricToBeCheck: String, lyrics: String) {
+        findViewById<TextView>(R.id.lyricMatchResult).text = if (lyrics.contains(lyricToBeCheck, ignoreCase = true)) "res: correct" else "res: too bad"
```

Comment:
The text is not cleared here after each song. But since it temporary, it is not so important I assume.